src/explore-education-statistics-common/src/components/ContentHtml.tsx

```diff
@@ -1,31 +1,107 @@
 import sanitizeHtml, { SanitizeHtmlOptions } from '@common/utils/sanitizeHtml';
+import parseHtmlString, { DOMNode, domToReact } from 'html-react-parser';
+import { Element } from 'domhandler/lib/node';
 import classNames from 'classnames';
-import React, { useMemo } from 'react';
+import React, { useMemo, useState } from 'react';
+import { GlossaryEntry } from '@common/services/types/glossary';
+import Modal from '@common/components/Modal';
+import styles from './ContentHtml.module.scss';
 
 export interface ContentHtmlProps {
   className?: string;
   html: string;
   sanitizeOptions?: SanitizeHtmlOptions;
   testId?: string;
+  getGlossaryEntry?: (slug: string) => Promise<GlossaryEntry>;
 }
 
 const ContentHtml = ({
   className,
   html,
   sanitizeOptions,
   testId,
+  getGlossaryEntry,
 }: ContentHtmlProps) => {
+  const [displayGlossaryEntry, setDisplayGlossaryEntry] = useState(false);
```

Comment:
I wouldn't bother with the `displayGlossaryEntry` flag if you don't need it.  Instead, you can make `glossaryEntry` "nullable", by replacing:

```
const [glossaryEntry, setGlossaryEntry] = useState({
    title: 'Glossary entry not retrieved',
    slug: '',
    body: '',
    link: '',
  });
```

with:

```
const [glossaryEntry, setGlossaryEntry] = useState<GlossaryEntry>();
```

Then check for it being defined or not rather than checking the flag.

Reply:
Unless you're doing it this way explicitly for caching purposes or some other reason?